examples/umami/common.rs

```diff
@@ -0,0 +1,761 @@
+    let re = Regex::new(&format!(r#"name="{}" value=['"](.*?)['"]"#, name)).unwrap();
+    match re.captures(&html) {
+        Some(value) => Some(value[1].to_string()),
+        None => None,
+    }
+}
+
+/// Anonymously load the contact form and POST feedback. The english boolean flag indicates
+/// whether to load the English form or the Spanish form.
+pub async fn anonymous_contact_form(user: &GooseUser, english: bool) -> GooseTaskResult {
+    let contact_form_url = if english {
+        "/en/contact"
+    } else {
+        "/es/contact"
+    };
+    let mut goose = user.get(contact_form_url).await?;
+
+    // We can't invoke common::validate_and_load_static_assets as while it's important
+    // to validate the page and load static elements, we then need to extra form elements
+    // from the HTML of the page. So we duplicate some of the logic, enhancing it for form
+    // processing.
+    let contact_form;
+    match goose.response {
+        Ok(response) => {
+            // Copy the headers so we have them for logging if there are errors.
+            let headers = &response.headers().clone();
+            match response.text().await {
+                Ok(html) => {
+                    // Be sure we've properly loaded the Contact form.
+                    let title = if english {
+                        "Website feedback"
+                    } else {
+                        "Comentarios sobre el sitio web"
+                    };
+                    if !valid_title(&html, title) {
+                        return user.set_failure(
+                            &format!("{}: title not found: {}", goose.request.url, title),
+                            &mut goose.request,
+                            Some(&headers),
+                            Some(&html),
+                        );
+                    }
+
+                    // Load all static elements on the page, as a real user would.
+                    load_static_elements(user, &html).await;
+
+                    // Scrape the HTML to get the values needed in order to POST to the
+                    // contact form.
+                    let form_build_id = get_form_value(&html, "form_build_id");
+                    if form_build_id.is_none() {
+                        return user.set_failure(
+                            &format!("{}: no form_build_id on page", goose.request.url),
+                            &mut goose.request,
+                            Some(&headers),
+                            Some(&html),
+                        );
+                    }
+
+                    // Build contact form parameters.
+                    // @TODO: dynamically generate the content.
+                    let params = [
+                        ("name", "@TODO: name"),
+                        ("mail", "nobody@example.com"),
+                        ("subject[0][value]", "@TODO: subject"),
+                        ("message[0][value]", "@TODO: message"),
```

Comment:
Is that hard to implement?

Reply:
Not at all, I just haven't defined what rules to follow when randomly generating the text.